### PR TITLE
Allow None as a value for a trait and ignore it when saving

### PR DESCRIPTION
**Pull request by zduey**

### CODE REVIEW

File: test.py [hunk 1]
```diff
@@ -28,6 +28,7 @@ def sample_recarray():
 class SomeSchema(Schema):
     x = Array(dtype=np.float)
     y = Array()
+    z = ArrayOrNone()
```
Comment (mivade) [inline test.py:31]: As we're adding more things to test, it is probably a good idea to name these things in a more readable way.

**Review by mivade — CHANGES_REQUESTED** [review]
You're missing a test of loading in cases where there was a None/no Nones.
